rpc/lib/client/http_json_client.go

```diff
@@ -298,16 +306,10 @@ func parseRemoteAddr(remoteAddr string) (network string, s string, err error) {
 	return protocol, address, nil
 }
 
-func makeErrorDialer(err error) func(string, string) (net.Conn, error) {
-	return func(_ string, _ string) (net.Conn, error) {
-		return nil, err
-	}
-}
-
-func makeHTTPDialer(remoteAddr string) func(string, string) (net.Conn, error) {
+func makeHTTPDialer(remoteAddr string) (func(string, string) (net.Conn, error), error) {
 	protocol, address, err := parseRemoteAddr(remoteAddr)
 	if err != nil {
-		return makeErrorDialer(err)
+		return nil, err
```

Comment:
wouldn't this result in panic when someone tries to call `nil` dialFn?

Reply:
Good catch, yes it would if the client using the code wouldn't follow the Golang convention and won't check the error before using the `dialFn`. Also, another Golang convention is to set each non-error returning value to a zero value when an error is returned. The zero value for an interface is nil. See the [Returning Errors Alongside Values](https://www.digitalocean.com/community/tutorials/handling-errors-in-go) section.

I removed the `makeErrorDialer` for clarity, it's better to have less code. But it's also possible that maybe I missed something (some other purpose why the `makeErrorDialer` was returned)...

Reply:
> some other purpose why the makeErrorDialer was returned

I doubt that